src/impl_methods.rs

```diff
@@ -2444,6 +2444,46 @@ where
         }
     }
 
+    /// Remove the `index`th elements along `axis` and shift down elements from higher indexes.
+    ///
+    /// Decreases the length of `axis` by one.
+    ///
+    /// ***Panics** if `axis` or `index` is out of bounds.
+    pub fn shift_remove_index(&mut self, axis: Axis, index: usize)
+    where
+        S: DataOwned + DataMut,
+    {
+        // TODO: It's possible to choose to shift the elment to the front or the back here,
+        // whichever is closer.
+        let (_, mut tail) = self.view_mut().split_at(axis, index);
+        // shift elements to the back
+        // use swap to keep all elements initialized (as required by owned storage)
+        Zip::from(tail.lanes_mut(axis)).for_each(|mut lane| {
+            let mut lane_iter = lane.iter_mut();
+            let mut dst = if let Some(dst) = lane_iter.next() { dst } else { return };
+
+            // Logically we do a circular swap here, all elements in a chain
+            // Using MaybeUninit to avoid unecessary writes in the safe swap solution
+            //
+            //  for elt in rest.iter_mut() {
+            //      std::mem::swap(dst, elt);
+            //      dst = elt;
+            //  }
+            //
+            let mut slot = MaybeUninit::<A>::uninit();
+            unsafe {
+                slot.as_mut_ptr().copy_from_nonoverlapping(dst, 1);
+                for elt in lane_iter {
```

Comment:
IMO, `src` would be a clearer name than `elt`.

Reply:
Hm matter of taste that I haven't addressed